### DISCUSSION TITLE
Postgresql multi column filter

### DISCUSSION THREAD

**Discussion by ghost** [dbgate/dbgate#817] [Q&A]
Hi there,

I've been using the multi-column filter feature extensively with MySQL and everything was fine, but it seems that it does not work correctly with PostgreSQL. When I search for a term, I encounter the following error: `operator does not exist: integer ~~* unknown (at line 7).`

Here is the code generated by the multi-column filter:

```
SELECT 
  "basetbl"."id" AS "id",
  "basetbl"."email" AS "email",
  "basetbl"."event" AS "event",
  "basetbl"."happenedAt" AS "happenedAt"
  FROM "public"."DraftBooking" "basetbl" 
WHERE (("basetbl"."id" ILIKE '%222%') OR ("basetbl"."email" ILIKE '%222%') OR ("basetbl"."event" ILIKE '%222%') OR ("basetbl"."happenedAt" ILIKE '%222%'))

ORDER BY "basetbl"."happenedAt" DESC
```

where "id" is an integer, "email" is a varchar, "event" is text, and "happenedAt" is a timestamp. I understand that it wants a cast for the "id" column, but I don't know if there is a way to fix that...

Thanks for your help!

**Comment by janproch** [comment]
This should be already fixed
